src/server/game/Spells/Spell.cpp

```diff
@@ -6555,9 +6555,8 @@ SpellCastResult Spell::CheckCast(bool strict, int32* param1 /*= nullptr*/, int32
                 if (m_originalCaster && m_originalCaster->GetTypeId() == TYPEID_PLAYER && m_originalCaster->IsAlive())
                 {
                     Battlefield* Bf = sBattlefieldMgr->GetBattlefieldToZoneId(m_originalCaster->GetMap(), m_originalCaster->GetZoneId());
-                    if (AreaTableEntry const* area = sAreaTableStore.LookupEntry(m_originalCaster->GetAreaId()))
-                        if (area->Flags[0] & AREA_FLAG_NO_FLY_ZONE  || (Bf && !Bf->CanFlyIn()))
-                            return SPELL_FAILED_NOT_HERE;
+                    if (Bf && !Bf->CanFlyIn())
```

Comment:
This check is about whether or not spell can be cast with SPELL_AURA_FLY & SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED.

Comment above this code metions to now allow casting fly spells if not required skills / allow always ghost flight spells.

I don't think its relevent in here tho, as mounts should be handled by mount capabilities (in Unit::UpdateMountCapabilities). And we have to allow fly auras from bosses etc.

Battlefield code stays untouched.

Reply:
Yes, this is legacy code from when SPELL_AURA_MOD_INCREASE_MOUNTED_FLIGHT_SPEED was a part of mount aura